modelerfour/flattener/flattener.ts

```diff
@@ -97,6 +97,14 @@ export class Flattener {
     schema.extensions = schema.extensions || {};
     schema.extensions[isCurrentlyFlattening] = true;
 
+    // ensure that parent schemas are done first -- this should remove 
+    // the problem when the order isn't just right.
+    for (const parent of values(schema.parents?.immediate)) {
```

Comment:
I'm guessing parent schemas were already being flattened before but this just forces it to happen first before children? The flattened parents are already marked as flattened so they won't be flattened again when encountered later?

Reply:
Yeah; coincidentally all the tests had the parents before the children, so this never came up before. Now it makes sure so that if something happens in the parent that would affect how the child would flatten, it's done early enough.